yt_dlp/extractor/espn.py

```diff
@@ -279,3 +275,52 @@ def _real_extract(self, url):
             'formats': formats,
             'subtitles': subtitles,
         }
+
+
+class WatchESPNIE(AdobePassIE):
+    _VALID_URL = r'https://www.espn.com/watch/player/_/id/(?P<id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})'
+    _TESTS = [{
+        'url': 'https://www.espn.com/watch/player/_/id/6eee6cb0-1795-49e3-84ea-766b517e0309',
+        'info_dict': {
+            'id': '6eee6cb0-1795-49e3-84ea-766b517e0309',
+            'ext': 'mp4',
+            'title': 'Miami vs. #2 Duke (M Basketball)',
+            'thumbnail': 'https://artwork.api.espn.com/artwork/collections/media/6eee6cb0-1795-49e3-84ea-766b517e0309/default?width=640&apikey=1ngjw23osgcis1i1vbj96lmfqs',
+            'duration': 7200
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        video_data = self._download_json(
+            'https://watch-cdn.product.api.espn.com/api/product/v3/watchespn/web/event?id=%s' % video_id, video_id)['page']['contents']
+        title = video_data['name']
+
+        if try_get(video_data, lambda x: x['streams'][0]['source']['name']) == 'ESPN+':
+            raise ExtractorError('ESPN+ streams are not currently supported', expected=True)
```

Comment:
Well passing cookies could work in theory, but it would require a completely different authentication process than what is used for the other streams. For example there is no equivalent or direct replacement for the `adobeToken` value that gets passed to the asset API call which returns the playlist URL. For ESPN+ streams to get the URL you need to make a call to `https://playback.svcs.plus.espn.com/media/***/scenarios/browser~ssai` with an authorization header, and I'm not sure how to get that header but after some research last night it looks like it could take a quite lengthy sequence of other API calls. Then you also need to get the authorization header to access the decryption key for the actual video. So ESPN+ is probably not impossible but would require a lot more time to figure out

Reply:
@ischmidt20 thank you for your work on this extractor as well as the other sports ones :)

I would like to ask that ESPN+ be included if possible, can make a formal site request if needed, but yt-dlp is able to download those streams if you pass it the correct information. Would be great to make it less manual if possible.

I've seen it posted about on youtube-dl [16359](https://github.com/ytdl-org/youtube-dl/issues/16359) as well as [reddit](https://www.reddit.com/r/youtubedl/comments/owsiay/tips_on_getting_espn_to_work_with_youtubedl/) but the process I've followed to manually do it is within browser using an authenticated account:

1. Grab the m3u8
2. Obtain the POST token response that comes from https://espn.api.edge.bamgrid.com/token
3. Using that access_token that is received to add it to the yt-dlp command headers using --add-header 'authorization:XXXX'